bin/install_cocoapods

```diff
@@ -13,13 +13,35 @@ LOCAL_CACHE_KEY="$BUILDKITE_PIPELINE_SLUG-local-pod-cache-$PODFILE_HASH"
 restore_cache "$LOCAL_CACHE_KEY"
 
 # If the `pod check` plugin is installed, use it to determine whether or not to install Pods at all
-# If it's not installed (or if it fails), we'll try to install Pods. 
+# If it's not installed (or if it fails), we'll try to install Pods.
 # If that fails, it may be due to an out-of-date repo. We can use `--repo-update` to try to resolve this automatically.
 if bundle exec pod plugins installed | grep -q check; then
 	bundle exec pod check || bundle exec pod install || bundle exec pod install --repo-update --verbose
 else
 	bundle exec pod install || bundle exec pod install --repo-update --verbose
 fi
 
+# Check that `Podfile.lock` was unchanged by `pod install`. If it was, it means
+# the lockfile might have been inadvertently changed.
+
+# `shasum` is available only on macOS
+if command -v shasum &> /dev/null; then
+  sha_command='shasum -a 256'
+else
+  sha_command='sha256sum'
+fi
```

Comment:
You'll notice that this block of code is duplicated from the `bin/hash_file` file above.

I started thinking about extracting this and updating all the `shasum`/`sha256sum` call sites that are still around, but decided to stop and submit this PR as focused only on:

1. Fixing the hashing behavior (currently a problem for PocketCasts iOS)
2. Adding the `Podfile.lock` check

I'd love to keep working on this, but I think it's better to ship this fixes first.

Reply:
Worth opening a GitHub issue to not forget about it?

Reply:
> Worth opening a GitHub issue to not forget about it?

Good idea, thanks. 👉 Created #21.

> why not call `hash_file` here...

Because in `hash_file` we use `sha*` to get the hash, but here we use it to check the hash matches (`$sha_command --check --status`).

I guess we could update `hash_file` to use more than just `$1` and so pass the options through, but that seems more in scope with #21.

Reply:
Ah, gotcha. Fair enough 👍